spec/controllers/campaigns_controller_spec.rb

```diff
@@ -123,4 +123,66 @@
       end
     end
   end
+
+  describe 'PATCH #publish' do
+    before do
+      patch :publish, params: { id: campaign.id }
+    end
+
+    context 'when state is new' do
+      it { expect(campaign).to have_state(:new) }
+
+      it { expect(campaign).to allow_event :publish }
+
+      it { expect(campaign).to_not allow_event :hold }
+
+      it { expect(campaign).to_not allow_event :close }
+    end
+
+    it { is_expected.to redirect_to(campaigns_path) }
+
+    it { is_expected.to set_flash }
+  end
+
+  describe 'PATCH #hold' do
+    before do
+      patch :publish, params: { id: campaign.id, aasm_state: campaign.publish }
+      patch :hold, params: { id: campaign.id, aasm_state: campaign.hold }
```

Comment:
Why do you perform two actions here?

Reply:
First one I need to perform transition from 'new' to 'published', second one from 'published' to 'on_hold'.